libs/database/src/workspace.rs

```diff
@@ -1227,3 +1227,164 @@ pub async fn update_comment_deletion_status<'a, E: Executor<'a, Database = Postg
 
   Ok(())
 }
+
+#[derive(PartialEq, Eq, Hash)]
+struct ReactionKey {
+  comment_id: Uuid,
+  reaction_type: String,
+}
+
+pub async fn select_reactions_for_published_view<'a, E: Executor<'a, Database = Postgres>>(
+  executor: E,
+  view_id: &Uuid,
+) -> Result<Vec<Reaction>, AppError> {
+  let rows = sqlx::query!(
+    r#"
+      SELECT
+        avr.comment_id,
+        avr.reaction_type,
+        au.uuid AS user_uuid,
+        au.name AS user_name
+      FROM af_published_view_reaction avr
+      INNER JOIN af_user au ON avr.created_by = au.uid
+      WHERE view_id = $1
+    "#,
+    view_id,
+  )
+  .fetch_all(executor)
+  .await?;
+  let reaction_to_users_map: HashMap<ReactionKey, Vec<AFWebUser>> = rows.iter().fold(
+    HashMap::new(),
+    |mut acc: HashMap<ReactionKey, Vec<AFWebUser>>, row| {
+      let users = acc
+        .entry(ReactionKey {
+          comment_id: row.comment_id,
+          reaction_type: row.reaction_type.clone(),
+        })
+        .or_default();
+      users.push(AFWebUser {
+        uid: row.user_uuid.clone(),
+        name: row.user_name.clone(),
+        avatar_url: None,
+      });
+      acc
+    },
+  );
+  let reactions = reaction_to_users_map
+    .iter()
+    .map(
+      |(
+        ReactionKey {
+          comment_id,
+          reaction_type,
+        },
+        users,
+      )| Reaction {
+        comment_id: *comment_id,
+        reaction_type: reaction_type.clone(),
+        react_users: users.clone(),
+      },
+    )
+    .collect();
+
+  Ok(reactions)
+}
+
+pub async fn select_reactions_for_comment<'a, E: Executor<'a, Database = Postgres>>(
+  executor: E,
+  comment_id: &Uuid,
+) -> Result<Vec<Reaction>, AppError> {
+  let rows = sqlx::query!(
```

Comment:
have you considered using `query_as!`? it might simplify what you're trying to do here

Reply:
query_as! would not be suitable in this case, because multiple rows in the query result are fold to a single Reaction, as oppose to one row to one Reaction mapping.

Reply:
@speed2exe Just realized that i could have simplified the query by using agg_array function. Though `query_as` is still a little tricky to implement, because i will need to add some sqlx macro on the entity (eg. AFWebUser), but the entity library currently doesn't have sqlx as a dependency, and i am not sure if i should pull in sqlx dependency for the crate just for this.

Will need to think more about this, `query_as` is certainly a cleaner solution if we can use it.